### PR TITLE
adding processingjob shape/validator

### PR DESCRIPTION
**Pull request by antgonza**
This PR adds two main changes to the ProcessingJob object:
- a shape property which returns the sample/metadata-columns related to the job, if possible
- adds a VALIDATOR job_type for qiita.processing_job_resource_allocation

### CODE REVIEW

**Comment by antgonza** [comment]
Sorry, forgot to add these images:
- List of jobs and their current usage/allocation, this was used to set the values in the SQL patch: Screen Shot 2020-06-24 at 7 24 24 AM
- alpha_vector is a clear outlier. As far as I remember, there was a bug in alpha group significance that made it use more memory than needed, thus the importance of grouping by job_number (time), note that the line is at 9.3G Screen Shot 2020-06-24 at 7 34 41 AM

File: qiita_db/processing_job.py [hunk 1]
```diff
@@ -1619,6 +1624,84 @@ def hide(self):
...
+                analysis_id = parameters['analysis']
+        elif self.command.software.name == 'Qiita':
+            if 'study' in parameters:
+                study_id = parameters['study']
+            elif 'study_id' in parameters:
+                study_id = parameters['study_id']
+            elif 'analysis' in parameters:
+                analysis_id = parameters['analysis']
+            elif 'analysis_id' in parameters:
+                analysis_id = parameters['analysis_id']
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_db/processing_job.py:1667]: Yup, joys of multiple developers:
- study: https://github.com/biocore/qiita/blob/master/qiita_ware/private_plugin.py#L195
- study_id: https://github.com/biocore/qiita/blob/master/qiita_ware/private_plugin.py#L406

**Comment by antgonza** [comment]
Thank you @ElDeveloper! Looking at the [codecov report](https://codecov.io/gh/biocore/qiita/pull/3012/changes); the only missing lines for tests are the `Qiita` block and the try/excepts. In the next PR: adding sample/column calculations; I actually needed to add more tests at least for the special case for `Qiita` ...

**Comment by ElDeveloper** [comment]
Thanks @antgonza!

**Pull request merged**